### DISCUSSION TITLE
What is the correct way to assign a custom score to messages coming from a SMTP-authenticated user

### DISCUSSION THREAD

**Discussion by wencywww** [rspamd/rspamd#4909] [Show and tell]
As we know, out of the box Rspamd does its checks on all messages reaching the MTA - regardles if they are coming from an external sender or from SMTP authenticated users as mentioned here:
https://rspamd.com/doc/faq.html#can-i-scan-outgoing-mail-with-rspamd

I'm using Rspamd together with Postfix via the milter protocol. The Postfix itself is configured to send to Rspamd the variable _{auth_authen}_ so Rspamd knows if the user is authenticated or not.
I want to be able to assign a custom (negative) score to these messages and after some investigations, i found this page:
https://rspamd.com/doc/configuration/settings.html#settings-structure

So, currently i'm using this:
Within _/etc/rspamd/local.d/settings.conf_ i added the following user "setting":
```
xxx_authenticated {
    priority = high;
    authenticated = yes;
    apply {
        XXX_AUTH_USER = -44.4;
        #groups_enabled = ["dkim_signing"]; # skip all checks but do the  dkim_signing
    }
    symbols [
        "XXX_AUTH_USER"
    ]
}
```
So, if i understood the things correct - this checks if the _authenticated_ flag is set and if so - it assigns to the message a symbol with a name of XXX_AUTH_USER and score of -44.4.
So far - so good, i can see in the logs that the symbol is really inserted (within the _rspamd_task_write_log:_ line), the score is applied and all seems to be OK, the symbol is visible within the WebGUI.

However, i'm unsure if this is the correct way to doing things like this and that's why i'm asking here.
Another concern is that i'm seeing this within the log, also:

```
 task; lua_task_insert_result_common: symbol insertion issue: unknown symbol CSM_AUTH_USER; trace: [1]:{/usr/share/rspamd/plugins/settings.lua:399 - fun [Lua]}; [2]:{/usr/share/rspamd/lualib/fun.lua:34 - call_if_not_empty [Lua]}; [3]:{/usr/share/rspamd/lualib/fun.lua:192
- each [Lua]}; [4]:{/usr/share/rspamd/plugins/settings.lua:398 - <unknown> [Lua]};
```
Regardles of this message - the symbol is inserted.

Moreover - using the commented _groups_enabled_ line i tried to switch off all other checks and only allow the Rspamd to do the DKIM signing. However, this does not work as expected - the checks are skipped indeed, but the signing is also missing.

I'm new to Rspamd, so - any ideas / suggestions?

**Comment by fatalbanana** [comment]
Maybe it suits to adjust thresholds rather than adding scoring.

Alternatively you might use a rule:

~~~lua
-- /etc/rspamd/rspamd.local.lua

rspamd_config.XXX_AUTH_USER = {
  callback = function(task)
    return (task:get_user() ~= nil)
  end,
  score = -44.4,
}
~~~

... Or you might try ignore the error.

RE: `dkim_signing`, there is not such a group, so you'd need to use something like `symbols_enabled = ["DKIM_SIGNED"]`
https://rspamd.com/doc/faq.html#can-i-just-sign-messages-using-dkim

**Reply by wencywww** [reply]
Thanks for answering so fast!

Well, i changed my settings.conf file like this:
```
xxx_authenticated {
    priority = high;
    authenticated = yes;
    apply {
        XXX_AUTH_USER = -44.4;
        symbols_enabled = ["DKIM_SIGNED"]; # skip all checks but do the dkim_signing
        flags = ["skip_process"]; # To skip message processing
    }
    symbols [
        "XXX_AUTH_USER"
    ]
}
```

Now the score is applied, the symbol is inserted and the signing is working. The warning 'symbol inserting' message is still there - but it is harmless according to you?

To be honest, i prefer to stay away from Lua for now - but what you consider the 'right way' for doing things like this - using rules with Lua ot using the approach i tried? Is there something incorrect/invalid/conceptual wrong with my way?

Finally - i'm afraid i was unable to understand the first sentence from your reply - what did you mena with "adjust thresholds rather than adding scoring" - can you post some example?

**Reply by fatalbanana** [reply]
If you'd be doing DKIM signing only there's no concern about score & the correct thing to do is not to insert the symbol.

Otherwise I'd suggest the correct thing to do is to rescore thresholds for`reject` & `add header` actions using settings.

**Reply by wencywww** [reply]
Well, in fact i wanted the symbol to be present, because the Web interface supports filtering by symbol name in its History section and it is useful to me. Otherwise - this will be enough, i suppose:
```
xxx_authenticated {
    authenticated = yes;
    apply {
        symbols_enabled = ["DKIM_SIGNED"]; # skip all checks but do the dkim_signing
        flags = ["skip_process"]; # To skip message processing
    }
 }
```

Interesting, what will be the score in this case - may be 0.00?
